addons/binding/org.openhab.binding.honeywellwifithermostat/src/main/java/org/openhab/binding/honeywellwifithermostat/internal/webapi/HoneywellWebsiteJetty.java

```diff
@@ -0,0 +1,236 @@
+package org.openhab.binding.honeywellwifithermostat.internal.webapi;
+
+import java.nio.charset.Charset;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.util.Fields;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.openhab.binding.honeywellwifithermostat.handler.honeywellWifiThermostatHandler;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatData;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatFanMode;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatSystemMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HoneywellWebsiteJetty {
+
+    private static HoneywellWebsiteJetty instance = null;
+    private Logger logger = LoggerFactory.getLogger(honeywellWifiThermostatHandler.class);
+
+    private String username;
+    private String password;
+
+    HttpClient httpclient;
+
+    protected HoneywellWebsiteJetty() {
+    }
+
+    public static HoneywellWebsiteJetty getInstance() {
+        if (instance == null) {
+            instance = new HoneywellWebsiteJetty();
+        }
+        return instance;
+    }
+
+    public void dispose() {
+        try {
+            httpclient.stop();
+        } catch (Exception e) {
+
+            e.printStackTrace();
+        }
+    }
+
+    public boolean isLoginValid() {
+        if (username == null || password == null) {
+            return false;
+        }
+        if (httpclient == null) {
+            tryLogin();
+        }
+        try {
+            ContentResponse cr = httpclient.GET("https://mytotalconnectcomfort.com/portal/Locations/");
+
+            if (cr.getStatus() == 500) {
+                return false;
+            }
+            if (cr.getStatus() == 401) {
+                // Try to login again.
+                tryLogin();
+                cr = httpclient.GET("https://mytotalconnectcomfort.com/portal/Locations/");
+
+            }
+            if (cr.getStatus() != 200) {
+                return false;
+            }
+
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+
+            e.printStackTrace();
+        }
+
+        return true;
+    }
+
+    private boolean tryLogin() {
+
+        SslContextFactory sslFactory = new SslContextFactory();
+
+        sslFactory.setExcludeProtocols(new String[] { "TLS", "TLSv1.2", "TLSv1.1" });
+
+        httpclient = new HttpClient(sslFactory);
```

Comment:
creating new instances of HttpClient is expensive. Maybe you can declare it as a field and reuse it?